### ISSUE TITLE
[Feature Request] Ability to provide specify format or generation of timestamps (ISO8601, specifically)

### ISSUE THREAD

**Issue by rex** [dynamoosejs/dynamoose#119]
I looked through all the code involved in timestamps and the issues involving timestamps and was a little shocked to learn that there is no way to specify the format of the timestamps. The schema defaults to `Date.now` but it would be absolutely incredible if we could specify something different, like `Date.toISOString`. A long string of numbers has no at-a-glance value whatsoever unless you are armed with a conversion calculator. Having ISO8601-compliant timestamps would add tremendous at-a-glance value to the data and eliminate the need for conversion to/from that format when interacting with the database.

I am interested to hear your thoughts, maybe there is something I missed in the process of that feature getting added!

**Edit:** Just wanted to let the record show that I'm aware how trivial the conversion of Unix timestamps to ISO8601 compliant timestamps is (see [related SO post](http://stackoverflow.com/a/12868272/439499)) but I am talking here about avoiding the whole thing in principle by moving the responsibility of that conversion to the ORM. 

**Edit 2:** I'm struggling to find a clean way of doing this and my options are pretty much zero. Tried an instance method, it strictly adheres to the type `Date`. Looked at virtual properties, zero documentation. The words `method`, `static`, and `virtual` are completely absent from the entire readme. So, this is actually becoming a cry for help.

**Comment by brandongoode** [comment]
First, sorry for the slow response (and sorry about the poor readme).

I think the ability to configure how dates are written and read would be a great add.  I originally chose the timestamp for its compactness and ease of sorting.  But it is a real pain to have to convert it every time I want to check something manually (I know I've done it a lot). 

I'll look into the option to configure it, but I am swamped and know it will be some time before I can do it. 

In the meantime, you could use the `set` and `get` options for an attribute to control how it is written to the DB.  

Something like:

```
  myDataAttr: {
    type: String,
    set: function(date) {
      if(date && date.toISOString) {
        return date.toISOString();
      }
    },
    get: function (val) {
      return new Date(val);
    }
  }
```

**Comment by rex** [comment]
Hello @brandongoode! No worries on either count, I got things worked out and ended up just reading through the pertinent parts of the code (which, incidentally, is a pretty great substitute for a readme).

In a stroke of luck on my end, the current use case I (and my team) have for dynamoose ended up just being a single field `updatedAt`, which I defined as follows:

```javascript
updatedAt: {
    type: String,
    default: Date.toISOString
}
```

And I simply set that field manually on instantiation of the model (`updatedAt: new Date().toISOString()`). 

I let my frustration get the better of my tone when I opened this issue, I didn't intend to sound so combative!
